source/val/validate_decorations.cpp

```diff
@@ -902,14 +902,15 @@ spv_result_t CheckDecorationsOfConversions(ValidationState_t& vstate) {
                     "Object operand of an OpStore.";
         }
 
-        const auto object_id = store->GetOperandAs<uint32_t>(1);
-        if (object_id != inst->id()) {
+        const auto object_id = store->word(use.second);
+        const auto ptr_id = store->GetOperandAs<uint32_t>(0);
+        if (object_id != inst->id() || ptr_id == inst->id()) {
```

Comment:
If the first part of this check fails its because our internal data structures are inconsistent. The use is tracking which word `inst->id()` will appear in. The check should really just be:
```
if (use.second != 2) {
 return error
}

const auto ptr_id = ...
```

Reply:
Oh ok. I updated it to check only the `use.second != 2`.